pkg/flow/tracing/wrap_tracer.go

```diff
@@ -41,29 +39,11 @@ type wrappedProvider struct {
 var _ trace.TracerProvider = (*wrappedProvider)(nil)
 
 func (wp *wrappedProvider) Tracer(name string, options ...trace.TracerOption) trace.Tracer {
-	innerTracer := wp.inner.Tracer(name, options...)
-	return &wrappedTracer{
-		inner:    innerTracer,
-		id:       wp.id,
-		spanName: wp.spanName,
+	// Inject the component ID as instrumentation scope attribute.
+	if wp.id != "" {
+		options = append(options, trace.WithInstrumentationAttributes(attribute.String(wp.spanName, wp.id)))
```

Comment:
Does the attribute include the component name? E.g. if the component is `otelcol.receiver.otlp "default"`, then would the instrumentation scope name be `otelcol.receiver.otlp.default` or `otelcol.receiver.otlp`? I think the convention is to not have the instance name, so then it would be `otelcol.receiver.otlp`. But I'm not sure what the Otel best practices are.

Reply:
Changing the attributes feels like a separate discussion to me, IMO. This PR does what #5285 asked for, and I'm wary about increasing the scope.

Reply:
That's ok, I'm happy for it to be merged as it is. Maybe it is even better that way, since it's more specific. If someone wants to narrow down to `otelcol.receiver.otlp.` they could probably use a TraceQL query to do it, and they could probably filter by `otelcol.receiver.` and `otelcol.` too.